Bottom: 23b5f14615cccfe3265deb777354f5d156a25f97
Top:    c0a9e90f8f28930dea43c199d3f2f795adcce80a
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-26 17:11:18 +0100

Refresh of remove-hack-in-draw-c-which-br

---

diff --git a/src/draw.c b/src/draw.c
index 831a993..0c3b45c 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -770,8 +770,7 @@ DrawLayerGroup (int group, const BoxType *drawn_area)
   int n_entries = PCB->LayerGroups.Number[group];
   Cardinal *layers = PCB->LayerGroups.Entries[group];
 
-  for (i = n_entries - 1; i >= 0;
-      i--, gui->set_layer (0, group, 0)) /* HACK: Subcomposite each layer in a layer group separately */
+  for (i = n_entries - 1; i >= 0; i--)
     {
       layernum = layers[i];
       Layer = PCB->Data->Layer + layers[i];
diff --git a/src/draw_funcs.c b/src/draw_funcs.c
index c1124d3..2dfe44b 100644
--- a/src/draw_funcs.c
+++ b/src/draw_funcs.c
@@ -334,10 +334,6 @@ draw_layer (LayerType *layer, void *userdata)
   /* print the non-clearing polys */
   r_search (layer->polygon_tree, drawn_area, NULL, poly_callback, &info);
 
-  /* HACK: Subcomposite polygons separately from other layer primitives */
-  /* Reset the compositing */
-  gui->set_layer (0, group, 0);
-
   if (TEST_FLAG (CHECKPLANESFLAG, PCB))
     return;
